examples/vnet/doca_dpdk.c

```diff
@@ -740,25 +787,53 @@ static void doca_dpdk_build_raw_data(uint8_t **header,
 
 	encap->conf.data = encap->data;
 	encap->conf.size = header - encap->data;
-	action->conf = &encap->conf;
+
+	if (hw_steering) {
+		action->type = RTE_FLOW_ACTION_TYPE_RAW_ENCAP;
+		action->conf = &encap->conf;
+	} else {
+		id = doca_encap_table_add_id(encap_data);
```

Comment:
I think you might have multi thread issue. 
two threads will can add_id
then get_refnt will return 2, and the ecanp won't be generated.

Reply:
I added a lock as you requested inside add_id, should be enough to avoid this case, what do you think ?

Reply:
this function call doca_encap_table_get_id and we lock inside doca_encap_table_get_id

Reply:
I changed the locking mechanism, such cases will not happen now.